src/analysis/normed_space/star.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2021 Frédéric Dupuis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Frédéric Dupuis
+-/
+
+import analysis.normed_space.linear_isometry
+
+/-!
+# Normed star rings and algebras
+
+A normed star ring is a star ring endowed with a norm such that the star operation is isometric.
+
+A C⋆-ring is a normed star ring that verifies the stronger condition `∥x⋆ * x∥ = ∥x∥^2` for all `x`.
+If a C⋆-ring is also a star algebra, then it is a C⋆-algebra.
+
+To get a C⋆-algebra `E` over field `𝕜`, use
+`[normed_field 𝕜] [star_ring 𝕜] [normed_ring E] [star_ring E] [cstar_ring E]
+ [normed_algebra 𝕜 E] [star_module 𝕜 E]`.
+
+-/
+
+local postfix `⋆`:1000 := star
+
+/-- A normed star ring is a star ring endowed with a norm such that `star` is isometric. -/
+class normed_star_ring (E : Type*) [normed_ring E] [star_ring E] :=
```

Comment:
Does this make sense as:
```suggestion
class normed_star_monoid (E : Type*) [normed_group E] [star_add_monoid E] :=
```
That's enough for `norm_star` to still be stated I think

Reply:
Indeed.